checkov/common/runners/runner_registry.py

```diff
@@ -577,16 +579,54 @@ def print_reports(
         exit_code = 1 if 1 in exit_codes else 0
         return cast(Literal[0, 1], exit_code)
 
+    # Define the function that will get the relay state from the Prisma Cloud Platform.
+    def get_sso_prismacloud_url(self, report_url: str, prisma_api_url: str, token: str) -> str:
+        url_saml_config = f"{prisma_api_url}/saml/config"
+        headers = {'x-redlock-auth': token}
+        try:
+            response = requests.get(url_saml_config, headers=headers, timeout=10)
+            response.raise_for_status()  # Raises an HTTPError if the response status is 4xx, 5xx
+            data = response.json()
+        except requests.exceptions.Timeout:
+            print("The request timed out")
+        except requests.exceptions.TooManyRedirects:
+            print("Too many redirects")
+        except requests.exceptions.RequestException as e:
+            print(f"Error while calling Prisma Cloud API: {e}")
+            return report_url
+
+        relay_state_param_name = data.get("relayStateParamName")
+        access_saml_url = data.get("redLockAccessSamlUrl")
+
+        if relay_state_param_name and access_saml_url:
+            parsed_url = urlparse(report_url)
+            uri = parsed_url.path
+            # If there are any query parameters, append them to the URI
+            if parsed_url.query:
+                uri = f"{uri}?{parsed_url.query}"
+            # Check if the URL already contains GET parameters.
+            if "?" in access_saml_url:
+                report_url = f"{access_saml_url}&{relay_state_param_name}={uri}"
+            else:
+                report_url = f"{access_saml_url}?{relay_state_param_name}={uri}"
+
+        return report_url
+
     def _print_to_console(self, output_formats: dict[str, str], output_format: str, output: str, url: str | None = None) -> None:
         """Prints the output to console, if needed"""
-
         output_dest = output_formats[output_format]
         if output_dest == CONSOLE_OUTPUT:
             del output_formats[output_format]
 
             print(output)
+
             if url:
+                if bc_integration.prisma_api_url is not None and bc_integration.get_auth_token() is not None:
+                    url = self.get_sso_prismacloud_url(report_url=url,
+                                                       prisma_api_url=bc_integration.prisma_api_url,
+                                                       token=bc_integration.get_auth_token())
```

Comment:
this is not the right place for it, because all related platform interaction is under `common.bridgecrew`. This would probably a good spot to add the function call https://github.com/bridgecrewio/checkov/blob/2df19723d80024544df918bf85cf73b88e5edec7/checkov/common/bridgecrew/platform_integration.py#L576

Reply:
This was integrated into the function get_sso_prismacloud_url moved to platform_integration.py